receiver/k8sobjectsreceiver/receiver.go

```diff
@@ -117,14 +117,21 @@ func (kr *k8sobjectsreceiver) startPull(ctx context.Context, config *K8sObjectsC
 	kr.stopperChanList = append(kr.stopperChanList, stopperChan)
 	kr.mu.Unlock()
 	ticker := NewTicker(config.Interval)
+	listOption := metav1.ListOptions{
+		FieldSelector: config.FieldSelector,
+		LabelSelector: config.LabelSelector,
+	}
+
+	if config.ResourceVersion != "" {
+		listOption.ResourceVersion = config.ResourceVersion
+		listOption.ResourceVersionMatch = metav1.ResourceVersionMatchNotOlderThan
```

Comment:
Why not strict? I can imagine a use case when a user wants to send objects to different pipelines depending on the version.

Reply:
I don't like making the interface more complicated and having different behavior between the operating modes. I think this option is already pretty niche. Usually, users shouldn't specify it and fetch all available versions by default. That's the default behavior if they don't specify the version, right?

My idea is that if users want to specify the version, they expect some specific schema that potentially can be different in future versions. Also, I think that the exact match is applied in the watch mode, right?

Reply:
@dmitryax yup, you're right the default behavior on watch mode is MatchExact, with that in mind I'm keeping the same behavior on pull mode.

If someday we get a new use case to specify the match strategy we can expose this through the config interface